### PR TITLE
Deal with different file encodings when submitting solutions

### PR DESCRIPTION
**Pull request by ambroff**
Fix for https://github.com/exercism/exercism.io/issues/2303

### CODE REVIEW

**Comment by kytrinyx** [comment]
Thanks!

Would you add the new dependencies to the .travis.yml build file?

- golang.org/x/net/html/charset
- golang.org/x/text/transform

**Comment by ambroff** [comment]
I actually did that here:
https://github.com/ambroff/cli/commit/ddbee5f4565a4b76ffad9ff090c0aa5ee579c194

Not sure why Travis didn't try the build again.

**Comment by kytrinyx** [comment]
Weird, those commits are not in this branch. Would you try doing a git push -f up to your branch to see if it re-triggers?

File: api/iteration.go [hunk 1]
```diff
@@ -73,3 +83,28 @@ func (iter *Iteration) isValidFilepath(path string) bool {
...
+	// Drop the UTF-8 BOM that may have been added. This isn't necessary, and
+	// it's going to be written into another UTF-8 buffer anyway once it's JSON
+	// serialized.
+	//
+	// The standard recommends omitting the BOM. See
+	// http://www.unicode.org/versions/Unicode5.0.0/ch02.pdf
+	decodedBytes = bytes.TrimPrefix(decodedBytes, utf8BOM)
+
+	s := string(decodedBytes)
+	return &s, nil
```
Comment (Tonkpils) [inline api/iteration.go]: Is there any reason why you're returning the pointer to the string here?

Comment (ambroff) [inline api/iteration.go]: A habit I picked up from the style guidelines of another project. The idea is that any function should either return a pointer to an initialized, usable object or a nil pointer and an error, but never an uninitialized object.

I don't feel strongly about this. Just habit.

**Comment by ambroff** [comment]
Welp, I can't figure out how to reopen this pull request now. The button is disabled.

**Comment by ambroff** [comment]
I gave up and recreated it here. https://github.com/exercism/cli/pull/184

**Comment by kytrinyx** [comment]
Ok, cool. Thanks!

**Pull request closed without merging**